shells/android/java/arcs/crdt/parcelables/ParcelableReferencable.kt

```diff
@@ -0,0 +1,75 @@
+package arcs.crdt.parcelables
+
+import android.os.Parcel
+import android.os.Parcelable
+import arcs.common.Referencable
+import arcs.data.RawEntity
+import java.lang.IllegalArgumentException
+import javax.annotation.OverridingMethodsMustInvokeSuper
+import kotlin.reflect.KClass
+
+/**
+ * Parcelable variant of the [Referencable] interface.
+ *
+ * All subclasses of [Referencable] need to have their own [Parcelable] implementation. [Type] is
+ * used to identify which subclass is being parceled.
+ */
+interface ParcelableReferencable : Parcelable {
+    val actual: Referencable
+
+    /** Indicates which subclass of [ParcelableReferencable] is being parceled. */
+    enum class Type(
+        val creator: Parcelable.Creator<out ParcelableReferencable>
+    ) : Parcelable {
+        // TODO: Add other ParcelableReferencable subclasses.
+        RawEntity(ParcelableRawEntity.CREATOR);
```

Comment:
Not a huge deal (ie. optional), since they're both in the same package, but: consider inverting this dependency by having the ReferencableParcelable class inject its creator rather than defining it here and having a circular dep.

Reply:
Ah, I think this may not be a good idea afterall, I like what you did with the interface's writeToParcel/CREATOR below. Very clean!